packages/binding-coap/src/coap-server.ts

```diff
@@ -337,387 +335,478 @@ export default class CoapServer implements ProtocolServer {
      */
     private async handleTdRequest(req: IncomingMessage, res: OutgoingMessage, thing: ExposedThing) {
         if (req.method !== "GET") {
-            res.code = "4.05";
-            res.end("Method Not Allowed");
+            this.sendMethodNotAllowedResponse(res);
             return;
         }
 
         const accept = req.headers.Accept;
 
         const contentSerdes = ContentSerdes.get();
 
-        if (accept == null || (typeof accept === "string" && contentSerdes.isSupported(accept))) {
-            debug(`Received an available or no Content-Format (${accept}) in Accept option.`);
-            const contentFormat = (accept as string) ?? ContentSerdes.TD;
-            res.setHeader("Content-Format", contentFormat);
-            res.code = "2.05";
+        const isUnsupportedAcceptValue = typeof accept === "string" && !contentSerdes.isSupported(accept);
 
-            const content = contentSerdes.valueToContent(thing.getThingDescription(), undefined, contentFormat);
-            const payload = await ProtocolHelpers.readStreamFully(content.body);
-            debug(`Sending CoAP response for TD with Content-Format ${contentFormat}.`);
-            res.end(payload);
-        } else {
+        if (isUnsupportedAcceptValue) {
             debug(`Request contained an accept option with value ${accept} which is not supported.`);
-            res.code = "4.06";
-            res.end(`Content-Format ${accept} is not supported by this resource.`);
+            this.sendErrorResponse(res, "4.06", `Content-Format ${accept} is not supported by this resource.`);
+            return;
         }
+
+        debug(`Received an available or no Content-Format (${accept}) in Accept option.`);
```

Comment:
```suggestion
        debug(`Request contained accept option with value ${accept} which is supported (or default/empty).`);
```

I struggled over this debug statement and it confused me. I tried to explain it better but I have difficulties to convey the message 🤷‍♂️